eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -15,32 +16,38 @@ jobs:
         VCPKG_DEFAULT_TRIPLET: 'x64-linux'
         CC: '/usr/bin/gcc-8'
         CXX: '/usr/bin/g++-8'
+        BuildArgs: '-j 10'
       Linux_x64_gcc9:
         OSVmImage: 'ubuntu-18.04'
         VcpkgInstall: 'curl[ssl] libxml2 openssl'
         VCPKG_DEFAULT_TRIPLET: 'x64-linux'
         CC: '/usr/bin/gcc-9'
         CXX: '/usr/bin/g++-9'
+        BuildArgs: '-j 10'
       Linux_x64:
         OSVmImage: 'ubuntu-18.04'
         VcpkgInstall: 'curl[ssl] libxml2 openssl'
         VCPKG_DEFAULT_TRIPLET: 'x64-linux'
+        BuildArgs: '-j 10'
       Win_x86:
         OSVmImage: 'windows-2019'
         VcpkgInstall: 'curl[winssl] libxml2'
         VCPKG_DEFAULT_TRIPLET: 'x86-windows-static'
         CMAKE_GENERATOR: 'Visual Studio 16 2019'
         CMAKE_GENERATOR_PLATFORM: Win32
+        BuildArgs: ''
```

Comment:
If you want to avoid setting these to the default empty value explicitly you can set them at a higher level like the job. and then exclude the duplication in the matrix entry. Any values in the specified in the matrix will be override the variables at the job level. See .NET as an example where we set some values in the globals.yml (https://github.com/Azure/azure-sdk-for-net/blob/master/eng/pipelines/templates/variables/globals.yml) and import it at the jobs level https://github.com/Azure/azure-sdk-for-net/blob/master/eng/pipelines/templates/jobs/archetype-sdk-client.yml#L108 so we don't have to repeat those default values.